packages/connect-conformance/src/web/transport.ts

```diff
@@ -0,0 +1,94 @@
+// Copyright 2021-2024 The Connect Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { createRegistry } from "@bufbuild/protobuf";
+import { ClientCompatRequest } from "../gen/connectrpc/conformance/v1/client_compat_pb.js";
+import {
+  Codec,
+  HTTPVersion,
+  Protocol,
+  Compression as ConformanceCompression,
+} from "../gen/connectrpc/conformance/v1/config_pb.js";
+import {
+  createConnectTransport,
+  createGrpcWebTransport,
+} from "@connectrpc/connect-web";
+import {
+  BidiStreamRequest,
+  ClientStreamRequest,
+  IdempotentUnaryRequest,
+  ServerStreamRequest,
+  UnaryRequest,
+} from "../gen/connectrpc/conformance/v1/service_pb.js";
+
+export function createTransport(req: ClientCompatRequest) {
+  let scheme = "http://";
+  if (req.serverTlsCert.length > 0) {
+    scheme = "https://";
+  }
+  const baseUrl = `${scheme}${req.host}:${req.port}`;
+
+  switch (req.httpVersion) {
+    case HTTPVersion.HTTP_VERSION_1:
+    case HTTPVersion.HTTP_VERSION_2:
+      break;
+    case HTTPVersion.HTTP_VERSION_3:
+      throw new Error("HTTP/2 and HTTP/3 are not supported");
```

Comment:
There's a mismatch between error message and code path here. 

I think we should be good to accept all three HTTP protocol versions here. Chrome and FF have supported h3 for a long time. https://caniuse.com/http3 Information about the state in Safari is muddy, but I see requests with h3 in the network inspector in 17.2.1, browsing to random sites.

Judging from the conformance runner output, we only run tests over plaintext HTTP/1.1. Is this because the runner configuration has `supportsH2c: false`?

Reply:
It is because we are still using rc1. If we upgrade to rc2 there are some Idempotency tests that fail for connect-node but they seem to be a problem of the runner, because using the latest runner we pass all the tests, so I refrained from upgrading until the next release.

Regarding `HTTP/3`, the latest runner actually even fails for `HTTP/1`, because it expects the client to use `HTTP/1`, even when the reference server supports `HTTP/2`. This I think should be changed, I am yet to report this.

Reply:
Okay, but please put a TODO at the H3 case.